packages/libs/react-ui/src/components/Accordion/AccordionSection.tsx

```diff
@@ -3,55 +3,53 @@
 import {
   accordionContentWrapperClass,
   accordionSectionClass,
+  accordionSectionHeadingClass,
   accordionTitleClass,
   accordionTitleVariants,
   toggleButtonClass,
 } from './Accordion.css';
 
 import { SystemIcon } from '@components/Icon';
 import classNames from 'classnames';
-import React, { FC, useEffect, useRef } from 'react';
+import type { FC } from 'react';
+import React from 'react';
 
 export interface IAccordionSectionProps {
-  title: React.ReactNode;
-  children: React.ReactNode;
-  isOpen: boolean;
-  onToggle: () => void;
-  onOpen?: () => void;
+  children?: React.ReactNode;
+  index?: number;
   onClose?: () => void;
+  onOpen?: () => void;
+  title: string;
+  onClick?: () => void;
+  openSections?: number[];
 }
 
 export const AccordionSection: FC<IAccordionSectionProps> = ({
-  isOpen = false,
-  title,
   children,
-  onToggle,
-  onOpen,
-  onClose,
+  index = 0,
+  title,
+  onClick,
+  openSections,
 }) => {
-  const didMountRef = useRef(false);
-
-  useEffect(() => {
-    if (!didMountRef.current) {
-      didMountRef.current = true;
-      return;
-    }
-
-    if (isOpen) onOpen?.();
-    else onClose?.();
-  }, [isOpen]);
-
+  const isOpen = openSections?.includes(index);
   return (
-    <div className={accordionSectionClass}>
+    <section
+      className={accordionSectionClass}
+      data-testid="kda-accordion-section"
+    >
       <div
-        data-testid="kda-accordion-title"
-        onClick={onToggle}
         className={classNames(
-          accordionTitleClass,
+          accordionSectionHeadingClass,
           accordionTitleVariants[isOpen ? 'opened' : 'closed'],
         )}
       >
-        <span>{title}</span>
+        <span
+          data-testid="kda-accordion-section-title"
+          onClick={onClick}
```

Comment:
I think the whole title + icon should be the button which should receive the onClick